Makefile & pre-commit hook instructions

sonnivs opened the pull request:
This PR adds the `pre-commit` library and a `pre-commit-hooks.yaml` file that configures the different steps that are run in the pre-commit hook.
Note: to enable the hook each person needs to run `poetry run pre-commit install` once to create the `.git/hooks/pre-commit` file automatically.

It adds a Makefile so we can easily run tests and auto-formatting. To use simply run `make <command>` in the terminal.




closes #12 and #22

JPhSachs commented:
@jp-sachs @JPhSachs Check whether `.git/hooks/pre-commit` can be explicitly included in the .gitignore file.

sonnivs commented:
I fixed the merge conflicts and updated as discussed to use `pre-commit` instead of our own hook. I also changed the PR description so the relevant information is easy to find in the future.
One important thing is that there is still one manual step that everyone has to do once to setup the pre-commit hook. 

So I think this is ready to merge @JPhSachs @jp-sachs

JPhSachs commented on .pre-commit-config.yaml:
I think there is no strong necessity for all of the comments, some could be removed IMHO.

sonnivs replied:
I reduced the amount and length of comments, let me know if this works for you

sonnivs replied:
I removed the unnecessary comments from .pre-commit-config.yaml, too and also pinned scikit-learn so we don't have to deal with the changes to check_estimator() now, and added https://github.com/hasso-plattner-institute/hfs/issues/35 so we can unpin that later. I dont want that to keep us from merging this PR.

JPhSachs reviewed (changes requested):
Looks good to me. I would only suggest minor changes for comments in .pre-commit-config.yaml
